elementary/monitor/data_monitoring/alerts/integrations/README.md

```diff
@@ -0,0 +1,128 @@
+# Integration contribution guide
+
+First of all, thank you for contributing a new integration to Elementary! We appreciate it :)
+This guide is meant to make this process easier for you and make the review process faster.
+
+Our integrations support either sending alerts, sending reports or both.
+Basically adding a new integration consists of adding the authentication params, implementing a client and formatting and sending the alerts.
+
+### CLI command and params
+
+The authentication params need to be added to the `edr monitor` CLI command params and passed on to the integration class.
+You can copy the behavior from the `slack_channel` and `slack_token` params.
+
+In order to support sending reports to this destination as well,
+please make sure those params are also available in the `edr send-report` CLI command.
+
+### Integration Client
+
+The client is the class that takes care of the authentication and all API calls that are required by the integration (listing channels, sending messages, etc.).
+Create the client class under the `elementary/clients` folder.
+The capabilities that should usually be supported in the client are:
+
+1. Authentication - usually in the `__init__` method.
+2. Listing channels / destinations / etc. - This is required to validate the destination exists and is accessible.
+3. Sending messages / triggering incidents - This is the main functionality of the client, and it will be used by the alert integration to send the alert.
+4. Sending files - This is required to support sending reports to this destination (to support `edr send report`).
+5. Error handling - This is required to handle authentication errors and other errors that might occur during the API calls.
+6. Searching users - This is required in messaging services in order to tag owners and subscribers in alerts
+
+### Formatting and sending alerts
+
+The integration class should be created under `elementary/monitor/data_monitoring/alerts/integrations/<integration_name>` folder and inherit from `BaseIntegration` (See slack.py for example).
+This class is responsible for formatting the alert and sending it to the integration, using the client class that was implemented earlier.
+
+_Note: There is a method called `get_integration_client` in DataMonitoringAlerts class that determines which alert integration to use based on the params passed to the `monitor` CLI command. Please make sure to add support for the new integration in this method._
```

Comment:
The method in `get_integration_client` in `DataMonitoringAlerts` using the method `Integrations.get_integration` and this is the one that needs to support the new integration (we don't want them to touch `DataMonitoringAlerts` 🙂